### DISCUSSION TITLE
How to use --config-setting to pass custom argument to setup.py when building

### DISCUSSION THREAD

**Discussion by LIONEFAN** [pypa/setuptools#3969] [Q&A] [answered]
Before, I use this command to pass arguments to setup.py to modify the wheel name and version suffix flexibility when building wheels.
`python setup.py bdist_wheel  --project_name my_test_project --weekly_build`

And my setyp.py use the following codes.
```
if '--project_name' in sys.argv:
     project_name_idx = sys.argv.index('--project_name')
     project_name = sys.argv[project_name_idx + 1]
     sys.argv.remove('--project_name')
     sys.argv.pop(project_name_idx)

if "--weekly_build" in sys.argv:
        VERSION = _VERSION + ".dev"
        sys.argv.remove("--weekly_build")
        project_name = project_name+"weekly"
```

It warning that avoid running ``setup.py`` directly. So I want to use the command `python -m build` .
I find it has a flag "--config-setting", and I use the command
`python -m build -w -n -C="--build-option=--project_name" -C="--build-option=test_project"
`refers to [this issue](https://github.com/pypa/build/issues/517)

But I don't know how to modify the codes in setup.py to receive the argument "project_name".
Can you help me to solve it?
And if you have a better idea to pass the arguments to setup.py when building, you can also tell me. Thanks!

**Comment by abravalheri** [comment, marked as answer]
Currently `--config-setting` does not connect in a direct manner with `setup.py`.
I suppose you can always use environment variables for that purpose.

**Reply by LIONEFAN** [reply]
Thanks your answer, I want to focus on the more standard and safe way to pass arguments. So which PR I should follow, Is [this one](https://github.com/pypa/setuptools/issues/2491)?

**Reply by abravalheri** [reply]
So far I would say the safest way is via env vars. You can follow that issue tracker, however enabling `--config-settings` is not going to automatically mean that any arbitrary arg will be passed to `setup.py`. What is more likely to happen is that we are going to select a few options and exposed only them.
